src/main/java/com/zextras/SlcwClient.java

```diff
@@ -129,4 +99,46 @@ public <T> OperationResult delete(T object) {
     LdapOperationExecutor executor = new LdapOperationExecutor(connection);
     return executor.executeDeleteOperation(entry);
   }
+
+  /**
+   * Gets an object from a record stored in the structure, otherwise throws an exception.
+   *
+   * @param id    a unique identifier that marks that particular record as unique from every other
+   *              record.
+   * @param clazz type of the class which represents an object that you wanted to get. (ex.
+   *              User.class)
+   * @param <T>   is a conventional letter that stands for "Type".
+   * @return an object of the given class.
+   */
+  public <T> T getById(String id, Class<T> clazz) {
+    T object = ObjectFactory.newObject(clazz);
+    SlcwEntry entry = new SlcwEntry(baseDn);
+    entry.getId().setPropertyValue(id);
+    mapper.map(object, entry);
+
+    LdapOperationExecutor executor = new LdapOperationExecutor(connection);
+    LdapOperationResult result = executor.executeGetOperation(entry);
+    entry.setAttributes(result.getEntries().get(0).getAttributes());
+
+    mapper.map(entry, object);
+    return object;
+  }
+
+  /**
+   * Counts the number of entries in the structure based on filter conditions.
+   *
+   * @param filter {@link Filter}.
+   * @return number of matching entries.
+   */
+  public long countBy(Filter filter) {
+    if (!(filter.getDn() == null) && !Objects.equals(filter.getDn(), baseDn)) {
```

Comment:
we can use opposite comparison check here :P
```suggestion
    if (filter.getDn() != null && !Objects.equals(filter.getDn(), baseDn)) {
```

Reply:
done!